lib/ts/recipe/totp/api/removeDevice.ts

```diff
@@ -0,0 +1,48 @@
+/* Copyright (c) 2023, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { send200Response } from "../../../utils";
+import STError from "../error";
+import { APIInterface, APIOptions } from "../types";
+import { makeDefaultUserContextFromAPI } from "../../../utils";
+
+import Session from "../../session";
+
+export default async function removeDevice(apiImplementation: APIInterface, options: APIOptions): Promise<boolean> {
+    if (apiImplementation.removeDevicePOST === undefined) {
+        return false;
+    }
+
+    let session = await Session.getSession(options.req, options.res);
```

Comment:
No. I think we shouldn't remove do it here. Removal of device should be possible only when you're fully logged in. Otherwise, someone can just remove all the devices after cracking the first factor.

Reply:
Okay. But then we need to remove all other validators, and just keep the allFactorsCompleted validator.

Reply:
> remove all other validators, and just keep the allFactorsCompleted validator

Hmm. Then this should happen in all TOTP APIs where we are overriding validators.

Reply:
TODO